hkmc2/shared/src/test/mlscript/lifter/FunInFun.mls

```diff
@@ -12,12 +12,17 @@ fun f(used1, unused1) =
   g(used2) + unused2
 f(1, 2)
 //│ JS (unsanitized):
-//│ let f, g$;
+//│ let g, f, g$;
 //│ g$ = function g$(used1, used2, g_arg) {
 //│   let used3;
 //│   used3 = 2;
 //│   return used1 + used2
 //│ };
+//│ g = function g(used1, used2) {
```

Comment:
It looks like these dead-code functions used to not be generated, so there seems to be a regression somewhere. Any idea?